dbms/src/Flash/Planner/plans/PhysicalAggregation.cpp

```diff
@@ -69,8 +69,11 @@ PhysicalPlanNodePtr PhysicalAggregation::build(
     analyzer.reset(aggregated_columns);
     analyzer.appendCastAfterAgg(expr_after_agg_actions, aggregation);
     /// project action after aggregation to remove useless columns.
-    const NamesAndTypes & schema = analyzer.getCurrentInputColumns();
-    expr_after_agg_actions->add(ExpressionAction::project(DB::toNames(schema)));
+    auto schema = PhysicalPlanHelper::addProjectAction(
+        expr_after_agg_actions,
+        analyzer.getCurrentInputColumns(),
```

Comment:
Why not pass analyzer into the function?

Reply:
Others LGTM

Reply:
> Why not pass analyzer into the function?

I mean if we pass analyzer into the function, the caller of the function will not consider what the before_schema is.

Reply:
Refactored to not use analyzer.